convert-pth-to-ggml.py

```diff
@@ -22,20 +22,34 @@
 import struct
 import numpy as np
 import torch
+import argparse
 
 from sentencepiece import SentencePieceProcessor
 
-if len(sys.argv) < 3:
-    print("Usage: convert-ckpt-to-ggml.py dir-model ftype\n")
-    print("  ftype == 0 -> float32")
-    print("  ftype == 1 -> float16")
-    sys.exit(1)
-
-# output in the same directory as the model
-dir_model = sys.argv[1]
-
-fname_hparams   = sys.argv[1] + "/params.json"
-fname_tokenizer = sys.argv[1] + "/../tokenizer.model"
+ARG_PARSER = argparse.ArgumentParser()
+ARG_PARSER.add_argument("--model",
+                        type=str,
+                        required=True,
+                        help="Model to convert")
+ARG_PARSER.add_argument("--ftype",
+                        type=str,
+                        required=True,
+                        choices=["f16", "f32"],
+                        help="Either f16 or f32")
+ARG_PARSER.add_argument("--output",
+                        type=str,
+                        required=True,
+                        help="Model to write")
+ARGS = ARG_PARSER.parse_args()
+
+FTYPE_IDX = -1
+if ARGS.ftype == "f16":
+    FTYPE_IDX = 1
+elif ARGS.ftype == "f32":
+    FTYPE_IDX = 0
```

Comment:
small suggestion:
```suggestion
FTYPE_IDX = {"f16": 1, "f32": 0}[ARGS.ftype]
```

Reply:
That's neat. I wanted to take another stab at it, but was stuck the error case (before I noticed I can use choose+require in argparse to guarantee the selection is total)